v-next/hardhat/test/internal/builtin-plugins/network-manager/hook-handlers/config.ts

```diff
@@ -604,6 +604,398 @@ describe("network-manager/hook-handlers/config", () => {
+
+            assert.equal(validationErrors.length, 0);
+          });
+
+          it("should allow valid private keys with missing hex prefix", async () => {
+            hardhatUserConfig.networks.localhost.accounts = [
+              "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+            ];
+            const validationErrors =
+              await validateUserConfig(hardhatUserConfig);
+            assert.equal(validationErrors.length, 0);
+          });
+        });
+
+        describe("not allowed values", () => {
+          describe("wrong private key formats", () => {
+            it("should not allow hex literals", async () => {
+              hardhatUserConfig.networks.localhost.accounts = [
+                0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,
+              ];
+
+              const validationErrors =
+                await validateUserConfig(hardhatUserConfig);
+
+              assert.notEqual(validationErrors.length, 0);
+              assert.equal(
+                validationErrors[0].message,
+                "Expected 'remote', an array of private keys, or an object with a mnemonic value and optional account details",
+              );
+            });
+
+            it("should not allow private keys of incorrect length", async () => {
+              hardhatUserConfig.networks.localhost.accounts = ["0xaaaa"];
+
+              let validationErrors =
+                await validateUserConfig(hardhatUserConfig);
+
+              assert.notEqual(validationErrors.length, 0);
+              assert.equal(
+                validationErrors[0].message,
+                "The private key must be a valid private key",
+              );
+              hardhatUserConfig.networks.localhost.accounts = [
+                "0xaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabb",
+              ];
+
+              validationErrors = await validateUserConfig(hardhatUserConfig);
+
+              assert.notEqual(validationErrors.length, 0);
+              assert.equal(
+                validationErrors[0].message,
+                "The private key must be a valid private key",
+              );
+            });
+
+            it("should not allow invalid private keys", async () => {
+              hardhatUserConfig.networks.localhost.accounts = [
+                "0xgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg",
+              ];
+
+              const validationErrors =
+                await validateUserConfig(hardhatUserConfig);
+
+              assert.notEqual(validationErrors.length, 0);
+              assert.equal(
+                validationErrors[0].message,
+                "The private key must be a valid private key",
+              );
+            });
+          });
+        });
+
+        it("should fail with invalid types", async () => {
+          const accountsValuesToTest = [123, [{}], { asd: 123 }];
```

Comment:
I grouped part of the V2 tests in a for loop to reduce the amount of repetitive code and improve overall readability. The same approach has been applied in other parts of the code as well